analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -19,23 +19,31 @@
  */
 
 using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Helpers;
 using SonarAnalyzer.SymbolicExecution.Constraints;
 
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
 {
     internal class ConstantCheck : SymbolicCheck
     {
         protected override ProgramState PreProcessSimple(SymbolicContext context) =>
-            GetConstraint(context.Operation.Instance) is { } constraint
+            ConstraintFromValue(context.Operation.Instance.ConstantValue.Value) is { } constraint
                 ? context.SetOperationConstraint(constraint)
                 : context.State;
 
-        private static SymbolicConstraint GetConstraint(IOperation operation) =>
```

Comment:
I suggest moving all these helper methods to a separate file with other helpers. They are not tightly coupled with the constant check.

Reply:
I'd keep them here, as their behavior is driven by what `ConstantCheck` is doing. And the other usage should only follow what the ConstantCheck it does.

Reply:
Only one of them should be public.